internal/cloudconfig/cloudinit/cloudinit_ubuntu.go

```diff
@@ -230,25 +237,31 @@ func (cfg *ubuntuCloudConfig) getCommandsForAddingPackages() ([]string, error) {
 	}
 
 	return cmds, nil
-
 }
 
-// addRequiredPackages is defined on the AdvancedPackagingConfig interface.
-func (cfg *ubuntuCloudConfig) addRequiredPackages() {
-	packages := []string{
-		"curl",
-		"tmux",
+func (cfg *ubuntuCloudConfig) getCommandsForAddingSnaps() ([]string, error) {
+	if len(cfg.Snaps()) == 0 {
+		return nil, nil
 	}
-	for _, pack := range packages {
-		cfg.AddPackage(pack)
+
+	var cmds []string
+	pkgCmder := cfg.paccmder[packaging.SnapPackageManager]
+
+	for _, snap := range cfg.Snaps() {
+		cmds = append(cmds, LogProgressCmd("Installing snap package: %s", snap))
+		cmds = append(cmds, pkgCmder.InstallCmd(snap))
 	}
+
+	return cmds, nil
 }
 
-var waitSnapSeeded = `
+// Wait for snap to seed, ensures that snapd is ready to be used.
+func (cfg *ubuntuCloudConfig) waitForSnap() string {
+	return `
 n=1
 while true; do
 
-echo "Attempt $n to wait for snapd to be seeded...\n"
+echo "Attempt $n to wait for snapd to be seeded..."
 snap wait core seed.loaded && break
 if [ $n -eq 5 ]; then
   echo "snapd not initialised"
```

Comment:
We really need to move away from this garbage. Cloud-init should be handling all this.. sigh. Next time we should make some bigger improvements here.